traffic_ops/traffic_ops_golang/cachegroupparameter/parameters.go

```diff
@@ -180,3 +185,120 @@ func (cgparam *TOCacheGroupParameter) Delete() (error, error, int) {
+		return
+	}
+	api.WriteResp(w, r, output)
+}
+
+func GetAllCacheGroupParameters(tx *sql.Tx) (tc.CacheGroupParametersList, error) {
+	parameters := map[string]string{
+		"orderby": "cachegroup",
+	}
+	// Query Parameters to Database Query column mappings
+	// see the fields mapped in the SQL query
+	queryParamsToQueryCols := map[string]dbhelpers.WhereColumnInfo{
+		"cachegroup":  dbhelpers.WhereColumnInfo{"cachegroup", api.IsInt},
+		"lastUpdated": dbhelpers.WhereColumnInfo{"last_updated", nil},
+		"parameter":   dbhelpers.WhereColumnInfo{"parameter", api.IsInt},
+	}
+	where, orderBy, pagination, _, errs := dbhelpers.BuildWhereAndOrderByAndPagination(parameters, queryParamsToQueryCols)
+	if len(errs) > 0 {
+		return tc.CacheGroupParametersList{}, util.JoinErrs(errs)
+	}
+
+	query := selectAllQuery() + where + orderBy + pagination
+	rows, err := tx.Query(query)
+	if err != nil {
+		return tc.CacheGroupParametersList{}, errors.New("querying cachegroupParameters: " + err.Error())
+	}
+	defer rows.Close()
+
+	paramsList := tc.CacheGroupParametersList{}
+	params := []tc.CacheGroupParametersNullable{}
+	for rows.Next() {
+		var p tc.CacheGroupParametersNullable
+		if err = rows.Scan(&p.CacheGroup, &p.Parameter, &p.LastUpdated); err != nil {
+			return tc.CacheGroupParametersList{}, errors.New("scanning cachegroupParameters: " + err.Error())
+		}
+		params = append(params, p)
+	}
+	paramsList.CacheGroupParameters = params
+	return paramsList, nil
+}
+
+func AddCacheGroupParameters(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	data, _ := ioutil.ReadAll(r.Body)
+	buf := ioutil.NopCloser(bytes.NewReader(data))
+
+	var paramsInt interface{}
+
+	decoder := json.NewDecoder(buf)
+	err := decoder.Decode(&paramsInt)
+	if err != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("parsing json: "+err.Error()), nil)
+		return
+	}
+
+	var params []tc.CacheGroupParametersNullable
+	_, ok := paramsInt.([]interface{})
+	var parseErr error = nil
+	if !ok {
+		var singleParam tc.CacheGroupParametersNullable
+		parseErr = json.Unmarshal(data, &singleParam)
+		if singleParam.CacheGroup == nil || singleParam.Parameter == nil {
+			api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("invalid cachegroup parameter."), nil)
+			return
+		}
+		params = append(params, singleParam)
+	} else {
+		parseErr = json.Unmarshal(data, &params)
+	}
+	if parseErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("parsing cachegroup parameter: "+parseErr.Error()), nil)
+		return
+	}
+
+	values := []string{}
+	for _, param := range params {
+		values = append(values, "("+strconv.Itoa(*param.CacheGroup)+", "+strconv.Itoa(*param.Parameter)+")")
+	}
+
+	insQuery := strings.Join(values, ", ")
+	_, err = inf.Tx.Tx.Query(insertQuery() + insQuery)
```

Comment:
you should be able (as long as our sqlx is recent enough) to do a bulk insert with a named query so you dont have to convert ints to strings and then joining them before adding them to the query. https://github.com/jmoiron/sqlx/pull/285 is when it was added

Reply:
doesnt look like ours is up to date enough to do this unfortunately